implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/DynatraceConfig.java

```diff
@@ -86,15 +110,22 @@ default DynatraceApiVersion apiVersion() {
                             if (apiVersionValidation.isValid()) {
                                 return checkAll(this,
                                         config -> {
-                                            if (config.apiVersion() ==  DynatraceApiVersion.V1) {
+                                            if (config.apiVersion() == DynatraceApiVersion.V1) {
                                                 return checkAll(this,
                                                         checkRequired("apiToken", DynatraceConfig::apiToken),
                                                         checkRequired("uri", DynatraceConfig::uri),
                                                         checkRequired("deviceId", DynatraceConfig::deviceId),
                                                         check("technologyType", DynatraceConfig::technologyType).andThen(Validated::nonBlank)
                                                 );
                                             } else {
-                                                return apiVersionValidation; // V2 validation comes here
+                                                return checkAll(this,
+                                                        c -> StepRegistryConfig.validate(c),
+                                                        check("apiToken", DynatraceConfig::apiToken)
```

Comment:
When an API token is present, we require both API token and URL. If there is no API token, we require only the URL, but i am not sure how to validate the URL in that case.

Reply:
If I interpret and simplify this, this means that the url must always be non-empty but the token can be empty, right?

Basically this is the truth table for the validation if I understand this correctly:
token | uri | valid
------------ | ------------- | -------------
0 | 0 | 0
0 | 1 | 1
1 | 0 | 0
1 | 1 | 1

Reply:
As I implemented this, I got confused since the exporter handles the case when the URL is empty, can we put this logic here?

Reply:
What do you think?
https://github.com/jonatan-ivanov/micrometer/commit/46f2b77b747b017ba37a0ef585b98d0dc65c3c02

Reply:
> if the token is present the url is required but if there is no token we fall back to a default url?

I think it is slightly different from defaulting to the OneAgent endpoint and making the token optional, since it will not produce an error in the case where only a token is passed, but no endpoint. I like your proposal a lot, but I would like to take care of the case where only a token is passed and inform the user. Might I propose something like this: https://github.com/dynatrace-oss-contrib/micrometer/commit/2c43bddc5e260d44dbc0f8f0b479e08fd18c3771 ? When implementing this I also realized that we had sent an empty token if none is set, so I added a fix while I was at it.